src/Sentry/Protocol/Transaction.cs

```diff
@@ -0,0 +1,327 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Json;
+using Sentry.Extensibility;
+using Sentry.Internal.Extensions;
+
+namespace Sentry.Protocol
+{
+    // https://develop.sentry.dev/sdk/event-payloads/transaction
+    /// <summary>
+    /// Sentry performance transaction.
+    /// </summary>
+    public class Transaction : ISpan, IScope, IJsonSerializable
+    {
+        private readonly IHub _hub;
+        private readonly SpanRecorder _spanRecorder = new SpanRecorder();
+
+        /// <inheritdoc />
+        public IScopeOptions? ScopeOptions { get; }
+
+        /// <summary>
+        /// Transaction name.
+        /// </summary>
+        public string Name { get; set; } = "unnamed";
+
+        /// <inheritdoc />
+        public SpanId SpanId
+        {
+            get => Contexts.Trace.SpanId;
+            private set => Contexts.Trace.SpanId = value;
+        }
+
+        /// <inheritdoc />
+        public SentryId TraceId
+        {
+            get => Contexts.Trace.TraceId;
+            private set => Contexts.Trace.TraceId = value;
+        }
+
+        /// <inheritdoc />
+        public DateTimeOffset StartTimestamp { get; private set; } = DateTimeOffset.UtcNow;
+
+        /// <inheritdoc />
+        public DateTimeOffset? EndTimestamp { get; private set; }
+
+        /// <inheritdoc />
+        public string Operation
+        {
+            get => Contexts.Trace.Operation;
+            internal set => Contexts.Trace.Operation = value;
+        }
+
+        /// <inheritdoc />
+        public string? Description { get; set; }
+
+        /// <inheritdoc />
+        public SpanStatus? Status
+        {
+            get => Contexts.Trace.Status;
+            private set => Contexts.Trace.Status = value;
+        }
+
+        /// <inheritdoc />
+        public bool IsSampled
+        {
+            get => Contexts.Trace.IsSampled;
+            set => Contexts.Trace.IsSampled = value;
+        }
+
+        /// <inheritdoc />
+        public SentryLevel? Level { get; set; }
```

Comment:
This probably doesn't make sense, right? What does "level" mean on a Transaction.

Reply:
Yeah it might be one of those fields that have no place on transaction. It would be really nice to have some "official" list.